### PR TITLE
Improve whilePresent description

### PR DESCRIPTION
**Pull request by jacwah**
See #386

### CODE REVIEW

**Comment by jacwah** [comment]
This conflicts with changes made in #433, solving the same issue.

File: cardshifter-core/src/main/java/com/cardshifter/core/groovy/WhilePresentDelegate.groovy [hunk 1]
```diff
@@ -50,7 +50,15 @@ class WhilePresentDelegate {
...
-                        'Give ' + deleg.description + ' ' + change + ' ' + resources.join(' and ')
+                        String changeStr
+                        if (change == 0) {
+                            changeStr = 'unchanged'
+                        }
+                        else {
+                            changeStr = [change.abs(), change < 0 ? 'less' : 'more'].join(' ')
+                        }
+
+                        [deleg.description.capitalize(), 'have', changeStr, resources.join(' and ')].join(' ')
```
Comment (Zomis) [inline cardshifter-core/src/main/java/com/cardshifter/core/groovy/WhilePresentDelegate.groovy]: This line and line 58 are a bit hard to understand (line 58 was very clever btw and slightly easier to read). Comments would help, perhaps with an example of the end result?

File: cardshifter-core/src/main/java/com/cardshifter/core/groovy/WhilePresentDelegate.groovy [hunk 2]
```diff
@@ -64,7 +72,7 @@ class WhilePresentDelegate {
             [withPriority: {int priority ->
                 [on: {Closure filter ->
                     addModifier(resources, priority, filter, { deleg ->
-                        'Set ' + resources.join(' and ') + ' to ' + deleg.description + ' ' + change
+                        [deleg.description.capitalize(), 'have', resources.join(' and '), 'set to', change].join(' ')
```
Comment (Zomis) [inline cardshifter-core/src/main/java/com/cardshifter/core/groovy/WhilePresentDelegate.groovy]: See https://github.com/Cardshifter/Cardshifter/pull/420/files#r58307727

**Comment by jacwah** [comment]
Fixed in #433

**Pull request closed without merging**